store/keystore.go

```diff
@@ -199,3 +211,36 @@ func saveToPem(out io.Writer, key openssl.PrivateKey) error {
 
 	return err
 }
+
+func loadPassphrase(passphrase_file string) (passphrase string, err error) {
+	var fi *os.File
+	var dat []byte
+	if passphrase_file != "" {
+		if passphrase_file == "-" {
+			log.Debugf("Read passphrase from stdin.")
+			fi, err = os.Open("/dev/stdin")
+			if err != nil {
+				log.Errorf("Failed to open stdin: %s", err)
+				return "", err
+			}
+		} else {
+			log.Debugf("Read passphrase from file.")
+			fi, err = os.Open(passphrase_file)
+			if err != nil {
+				log.Errorf("Failed to open passphrase file: %s", err)
+				return "", err
+			}
+		}
+		dat, err = ioutil.ReadAll(fi)
+		if err != nil {
+			log.Errorf("Failed to read passphrase: %s", err)
+			return "", err
+		}
+		passphrase = strings.TrimRight(string(dat), "\n")
+		fi.Close()
```

Comment:
With the new error handling, this should be moved. Instead of calling it here, call it right before `ReadAll`, like this:

```go
defer fi.Close()
dat, err = ioutil.ReadAll(fi)
```

Then it will be closed when the function exits, regardless of where it exits. This way it will also be called when exiting through the error return.